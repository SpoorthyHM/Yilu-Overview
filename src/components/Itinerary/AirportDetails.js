import React from 'react';

const airportDetails =(props) =>{
    const details = props.segment;
    let statusStyle = 'Green';
    if(details.detail.status === "Busy") {
        statusStyle =  {
            color: 'orange'
          };
    }
    return (
        <div>
            <p>{details.origin.value} {details.origin.type}</p>
            <p>{details.detail.terminal}<span style={statusStyle} className="status">{details.detail.status}</span></p>
        </div>
    )
}

export default airportDetails;