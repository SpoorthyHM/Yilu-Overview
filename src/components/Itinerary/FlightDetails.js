import React from 'react';

const flightDetails =(props) =>{
    const details = props.segment;
    let flightNumber = details.detail.flight_number;
    let boardingTime =  new Date( details.detail.boarding *1000).toLocaleString().split(",")[1];
    let origin = details.origin.value;
    let destination = details.destination.value;
    let statusStyle = {};
    if(details.detail.status === 'Ontime') {
        statusStyle = {
            backgroundColor: 'green'
        };
    }
    return (
        <React.Fragment>
            <div className="card-header bg-info text-white">
                <p>{flightNumber}<span className="status">{boardingTime}</span></p>
            </div>
            <div className="card-body">
                <p>{origin}
                    <span className="status">{destination}</span>
                </p>
            </div>
            <div className="card-footer bg-light">
                <p><span>Gate: {details.detail.gate}</span>
                    <span style={{marginLeft: '10px'}}>Seat: {details.detail.seat}</span>
                    <span style={statusStyle} className="status">{details.detail.status}</span>
                </p>
            </div>
        </React.Fragment>
    )
}

export default flightDetails;