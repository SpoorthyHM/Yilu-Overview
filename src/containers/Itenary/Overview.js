import React, {Component} from 'react';
import Layout from '../../hoc/Layout/Layout';
import axios from '../../axios-itenary';
import Itinerary from '../../components/Itinerary/Itinerary';

class Overview extends Component {

    state = {
        itenaryDetails: {}
    }

    initItenaryDetails = () => {
        
        axios.get('/itenary')
        .then(response => {
            this.setState({itenaryDetails: response.data});
        })
        .catch(error => {
            console.log('Error');
        });
    }

    componentDidMount() {
        this.initItenaryDetails();
    }
    
    render(){
        var itenary = {};
        if(this.state.itenaryDetails) {
            itenary = this.state.itenaryDetails.itenaryDetails;
        }
        
        return (
            <Layout>
                <div className="card">
                    <div className="card-header bg-primary text-white">Overview</div>
                    <div className="card-body bg-light text-dark">
                        Berlin - Munchen
                                <Itinerary details={itenary}/>
                    </div> 
                </div>                                             
            </Layout>
        );
    }
}

export default Overview;