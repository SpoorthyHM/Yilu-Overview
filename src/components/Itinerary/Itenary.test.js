import React from 'react';
import { shallow, mount } from 'enzyme';
import Itenary from './Itinerary';
import FlightDetails from './FlightDetails';
import AirportDetails from './AirportDetails';

describe('<Itenary />', () => {
    let details = {};
    const itenaryDetails = {
        "itinerary_id": "3dd59474-f467-42bc-8098-3c1695ab362d",
        "traveler": {
            "first_name": "John",
            "middle_name": "C.",
            "last_name": "Doe"
        },
        "origin_iata": "TXL",
        "destination_iata": "MUC",
        "segments": [{
                "segment_id": "452dfb51-0ac6-4274-b740-af6f89bc6116",
                "type": "GROUND",
                "origin": {
                    "value": "Alte Schönhauser Straße 3, 10119 Berlin",
                    "type": "STREET_ADDRESS"
                },
                "departure": 1523596800,
                "destination": {
                "type": "AIRPORT",
                "value": "TXL"
            }
        },
        {
            "segment_id": "39552e6d-6785-4e83-82dc-157a45773b45",
            "type": "AIRPORT",
            "origin": {
            "type": "AIRPORT",
            "value": "TXL"
            },
                "departure": 1523599200,
                "detail": {
                    "busyness_index": 10,
                    "terminal" : "Terminal C",
                    "status" : "Busy"
                }
        }]};

    it('should render correctly with empty props', () => {
        const component = shallow(<Itenary {...details}/>);
        component.setProps(details);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with props', () => {
        const component = shallow(<Itenary {...itenaryDetails}/>);
        component.setProps(itenaryDetails);
        expect(component).toMatchSnapshot();
      });

      it('should test whether <AirportDetails> is rendered when there is data', () => {
        const component = mount(<Itenary {...itenaryDetails}/>);
        component.setProps({details: itenaryDetails});
        expect(component.find(AirportDetails)).toHaveLength(1);
        expect(component.find(FlightDetails)).toHaveLength(0);
      });
  
})