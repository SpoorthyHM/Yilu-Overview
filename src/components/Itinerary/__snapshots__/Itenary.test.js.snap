// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Itenary /> should render correctly with empty props 1`] = `ShallowWrapper {}`;

exports[`<Itenary /> should render correctly with props 1`] = `ShallowWrapper {}`;
